File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2864-maximum-odd-binary-number/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2864-maximum-odd-binary-number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int n = s.length();
        int count_one = 0;
        
        for(char ch : s) 
            if(ch == '1') 
                count_one++;
        
        int count_zero = n - count_one;

        
        
//         String ans = "";
//         for(int i=0;i<count_one - 1;i++) ans += "1";
//         for(int i=0;i<count_zero;i++) ans += "0";
//         if(count_one > 0) ans += "1";
        
//         return ans;
        
        for(int i = 0; i < count_one - 1;i++) s[i] = '1';
        for(int i = count_one - 1;i < n - 1 ; i++) s[i] = '0';
        s[n-1] =  '1';
        
        return s;
    }
};

File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2864-maximum-odd-binary-number/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2864-maximum-odd-binary-number.java
public class Solution {
    public String maximumOddBinaryNumber(String s) {
        int n = s.length();
        int countOne = 0;

        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == '1') {
                countOne++;
            }
        }

        int countZero = n - countOne;

        // Create a StringBuilder to modify the string in place
        StringBuilder modifiedString = new StringBuilder(s);

        // Set '1's for the first countOne - 1 characters
        for (int i = 0; i < countOne - 1; i++) {
            modifiedString.setCharAt(i, '1');
        }

        // Set '0's for the next countZero characters
        for (int i = countOne - 1; i < n - 1; i++) {
            modifiedString.setCharAt(i, '0');
        }

        // Set the last character to '1'
        modifiedString.setCharAt(n - 1, '1');

        return modifiedString.toString();
    }
}
